DependencyInjection/Configuration.php

```diff
@@ -41,24 +42,80 @@ public function getConfigTreeBuilder() : TreeBuilder
 
         $rootNode
             ->children()
-                ->scalarNode('dir_name')->defaultValue('%kernel.root_dir%/DoctrineMigrations')->cannotBeEmpty()->end()
-                ->scalarNode('namespace')->defaultValue('Application\Migrations')->cannotBeEmpty()->end()
-                ->scalarNode('table_name')->defaultValue('migration_versions')->cannotBeEmpty()->end()
-                ->scalarNode('column_name')->defaultValue('version')->end()
-                ->scalarNode('column_length')->defaultValue(14)->end()
-                ->scalarNode('executed_at_column_name')->defaultValue('executed_at')->end()
-                ->scalarNode('all_or_nothing')->defaultValue(false)->end()
                 ->scalarNode('name')->defaultValue('Application Migrations')->end()
-                ->scalarNode('custom_template')->defaultValue(null)->end()
-                ->scalarNode('organize_migrations')->defaultValue(false)
+                ->arrayNode('migrations_paths')
```

Comment:
Note: once this is merged, we should also allow for using the new format in the current major version for forwards compatibility. We can take care of that separately.